libs/npm/core/src/utils/elastic/generateQueryShouldValues.ts

```diff
@@ -0,0 +1,14 @@
+import { IFilterSettingsModel } from '../../hooks';
+import { generateMatch } from './generateMatch';
+/**
+ * Generates an Elasticsearch query based on specified 'query'.
+ * @param settings Form values that will be used to configure the elasticsearch query.
+ * @param query Original query object.
+ * @returns Elasticsearch query JSON.
+ */
+export const generateQueryShouldValues = (
+  settings: Omit<IFilterSettingsModel, 'size' | 'searchUnpublished'>,
+): any[] => {
+  const matches = generateMatch('byline', settings.contributorNames);
```

Comment:
This isn't going to work correctly.  I reviewed this morning.

The `content.byline` and `content.contributor.name` is not always going to be the same.  We are adding code to apply the contributor based on the byline, but this isn't the inverse also.  If an editor selects the contributor we don't apply it to the byline, nor do we currently want to.

If an Advanced Search selects a contributor you will need to search the `content.byline` for the name, and the `content.contributorId` for the selected contributor.  If they only select one contributor it must find them and will be an AND.  If they select multiple, then it will be an OR/SHOULD.

Also when the user has selected the Byline checkbox to search for keywords,  you will need to search the `content.contributor.name` for any of the keywords.

Reply:
I changed the logic to cover your comments. Please take a look. Thank you!